radio/src/model_init.cpp

```diff
@@ -0,0 +1,144 @@
+
+void clearInputs()
+{
+  memset(g_model.expoData, 0, sizeof(g_model.expoData));
+}
+
+void defaultInputs()
+{
+  for (int i=0; i<NUM_STICKS; i++) {
+    uint8_t stick_index = channelOrder(i+1);
+    ExpoData *expo = expoAddress(i);
+    expo->srcRaw = MIXSRC_Rud - 1 + stick_index;
+    expo->curve.type = CURVE_REF_EXPO;
+    expo->chn = i;
+    expo->weight = 100;
+    expo->mode = 3; // TODO constant
+    for (int c = 0; c < 3; c++) {
+      g_model.inputNames[i][c] = STR_VSRCRAW[2 + 4 * stick_index + c];
+    }
+#if LEN_INPUT_NAME > 3
+    g_model.inputNames[i][3] = '\0';
+#endif
+  }
+  storageDirty(EE_MODEL);
+}
+
+void clearMixers()
+{
+  memset(g_model.mixData, 0, sizeof(g_model.mixData));
+}
+
+void defaultMixers()
+{
+  for (int i=0; i<NUM_STICKS; i++) {
+    MixData * mix = mixAddress(i);
+    mix->destCh = i;
+    mix->weight = 100;
+    mix->srcRaw = i+1;
+  }
+  storageDirty(EE_MODEL);
+}
+
+void defaultModelRegistrationID()
+{
+  memcpy(g_model.modelRegistrationID, g_eeGeneral.ownerRegistrationID, PXX2_LEN_REGISTRATION_ID);
+}
+
+void defaultGVars()
+{
+#if defined(FLIGHT_MODES) && defined(GVARS)
+  for (int fmIdx = 1; fmIdx < MAX_FLIGHT_MODES; fmIdx++) {
+    for (int gvarIdx = 0; gvarIdx < MAX_GVARS; gvarIdx++) {
+      g_model.flightModeData[fmIdx].gvars[gvarIdx] = GVAR_MAX + 1;
+    }
+  }
+#endif
+}
+
+void setVendorSpecificModelDefaults()
+{
+#if defined(FRSKY_RELEASE)
+  g_model.moduleData[INTERNAL_MODULE].type = IS_PXX2_INTERNAL_ENABLED() ? MODULE_TYPE_ISRM_PXX2 : MODULE_TYPE_XJT_PXX1;
+  g_model.moduleData[INTERNAL_MODULE].channelsCount = defaultModuleChannels_M8(INTERNAL_MODULE);
+  #if defined(EEPROM)
+    g_model.header.modelId[INTERNAL_MODULE] = findNextUnusedModelId(id, INTERNAL_MODULE);
+    modelHeaders[id].modelId[INTERNAL_MODULE] = g_model.header.modelId[INTERNAL_MODULE];
+  #endif
+#endif
+
+    // TODO: we should probably have some default trainer mode
+    //       per radio, depending on what is supported
+    //
+#if defined(PCBXLITE)
+  g_model.trainerData.mode = TRAINER_MODE_MASTER_BLUETOOTH;
+#endif
+
+#if defined(RADIOMASTER_RTF_RELEASE)
+  // Those settings are for headless radio
+  g_model.trainerData.mode = TRAINER_MODE_SLAVE;
+  g_model.moduleData[INTERNAL_MODULE].type = MODULE_TYPE_MULTIMODULE;
+  g_model.moduleData[INTERNAL_MODULE].setMultiProtocol(MODULE_SUBTYPE_MULTI_FRSKY);
+  g_model.moduleData[INTERNAL_MODULE].subType = MM_RF_FRSKY_SUBTYPE_D8;
+  g_model.moduleData[INTERNAL_MODULE].failsafeMode = FAILSAFE_NOPULSES;
+#endif
+}
+
+void applyDefaultTemplate()
+{
+  defaultInputs();
+  defaultMixers();
+  defaultGVars();
+
+  defaultModelRegistrationID();
+
+#if defined(COLORLCD)
+  //TODO: not sure yet we need it here
+  loadDefaultLayout();
+
+  // enable switch warnings
+  for (int i = 0; i < NUM_SWITCHES; i++) {
+    g_model.switchWarningState |= (1 << (3*i));
+  }
+#endif
+
+  // TODO: what about switch warnings in non-color LCD radios?
+
+  setVendorSpecificModelDefaults();
```

Comment:
move in modelDefault() after applyDefaultTemplate() for easier understanding ?

Reply:
Why not, I have no preferences here.